python/ray/tune/trial.py

```diff
@@ -443,6 +445,23 @@ def last_result(self) -> dict:
     def last_result(self, val: dict):
         self._last_result = val
 
+    @property
+    def logdir(self):
+        if not self.relative_logdir:
+            return None
+        return str(Path(self.local_dir).joinpath(self.relative_logdir))
+
+    @logdir.setter
+    def logdir(self, logdir):
+        relative_logdir = Path(logdir).relative_to(self.local_dir)
+        if ".." in str(relative_logdir):
+            raise ValueError("logdir must not contain `..`")
```

Comment:
Let's raise a more actionable error message here:

```suggestion
            raise ValueError(
                f"The `logdir` points to a directory outside the trial's `local_dir` "
                f"({self.local_dir}), which is unsupported. Use a logdir within the "
                f"local directory instead. Got: {logdir}"
            )
```

Reply:
Done